homeassistant/components/wsdot/sensor.py

```diff
@@ -44,22 +49,46 @@
 async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Migrate a platform-style wsdot to entry-style."""
+    # old-style config was entered by hand. make sure all values are valid
+    api_key = config[CONF_API_KEY]
+    session = async_get_clientsession(hass)
+    try:
+        # check for valid API Key
+        wsdot_api.WsdotTravelTimes(
+            api_key=api_key, session=session
+        ).get_all_travel_times()
+    except wsdot_api.WsdotTravelError as wsdot_error:
+        raise PlatformNotReady from wsdot_error
+
+    for old_entry in hass.config_entries.async_loaded_entries(DOMAIN):
+        await hass.config_entries.async_remove(old_entry.entry_id)
+    await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
     """Set up the WSDOT sensor."""
     sensors = []
     session = async_get_clientsession(hass)
-    api_key = config[CONF_API_KEY]
-    wsdot_travel = WsdotTravelTimes(api_key=api_key, session=session)
-    for travel_time in config[CONF_TRAVEL_TIMES]:
+    api_key = entry.data[CONF_API_KEY]
+    wsdot_travel = wsdot_api.WsdotTravelTimes(api_key=api_key, session=session)
```

Comment:
this is something that should exist in async_setup_entry in init.py

Reply:
Are you asking that `WsdotTravelTimes` be instantiated in `__init__.async_setup_entry`? I don't know why, the class is only useful to `WashingtonStateTravelTimeSensor` instances.